### PR TITLE
Dribble stats

### PR DESCRIPTION
**Pull request by dtracers**
Add stats for cars that can dribble

### CODE REVIEW

File: carball/analysis/events/carry_detection.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    def get_distance(self, player_frames: pd.DataFrame, start: int, end: int) -> pd.DataFrame:
+        x1 = player_frames.pos_x[start]
+        y1 = player_frames.pos_y[start]
+
+        x2 = player_frames.pos_x[end]
+        y2 = player_frames.pos_y[end]
+
+        return ((x2 - x1) ** 2 + (y2 - y1) ** 2) ** 0.5
+
+    def creat_start_end_frames(self, carry_frames: pd.DataFrame) -> Tuple[List[int], List[int]]:
```
Comment (twobackfromtheend) [inline carball/analysis/events/carry_detection.py]: Apart from the typo in the function name, what're you trying to do here w/ start frames and end_frames? It feels like there has to be a better way.

File: carball/analysis/events/carry_detection.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+            merged_start.append(previous_start)
+            merged_end.append(previous_end)
+        player_carry_data.start_frames = merged_start
+        player_carry_data.end_frames = merged_end
+
+    def expand_carries(self, carry_data: CarryData, player_carry_data: CarryData,
+                       player: Player, proto_game: game_pb2.Game) -> CarryData:
+        """
+        Expands / merges carries based on hit list.
+        If nothing is modified then this returns valid_frames
```
Comment (twobackfromtheend) [inline carball/analysis/events/carry_detection.py]: This docstring confuses me more than it explains stuff. What's "expands"? What's valid_frames?

File: carball/analysis/events/carry_detection.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        start_frames = player_carry_data.start_frames
+        end_frames = player_carry_data.end_frames
+
+        merged_start = []
+        merged_end = []
+
+        previous_start = -1
+        previous_end = 1
+
+        for frame_index in range(len(start_frames) - 1):
```
Comment (twobackfromtheend) [inline carball/analysis/events/carry_detection.py]: There's a lot going on here with not much explanation. The docstring doesn't help much as it doesn't say what a valid dribble set is or what separated dribbles are. 
It's also hard to follow the logic within here (as I have no clue how it's trying to achieve what I think it's trying to achieve).

File: carball/analysis/events/carry_detection.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+            total_time = carry_frames.game.delta.sum()
+            if total_time < 1:
+                continue
+
+            ball_carry = proto_game.game_stats.ball_carries.add()
+            ball_carry.start_frame_number = start_frames[i]
+            ball_carry.end_frame_number = end_frames[i]
+            ball_carry.player_id.id = player.id.id
+            ball_carry.carry_time = total_time
+            ball_carry.distance_traveled = self.get_distance(player_frames, start_frames[i], end_frames[i])
```
Comment (twobackfromtheend) [inline carball/analysis/events/carry_detection.py]: Why don't you just sum up the ball position diffs?
`(ball.x.diff()**2 + ball.y.diff()**2).sqrt()` or something?

Comment (dtracers) [reply carball/analysis/events/carry_detection.py]: This is straight line distance not along the path difference.
Though I might add that with player too!
Good suggestion

File: carball/analysis/events/carry_detection.py (hunk 1 above)
Comment (dtracers) [reply carball/analysis/events/carry_detection.py]: I am trying to get the first frame of the dribble and the last frame of the dribble which is then very useful later on

Comment (twobackfromtheend) [reply carball/analysis/events/carry_detection.py]: There's a bit of a random threshold of 3.

Comment (dtracers) [reply carball/analysis/events/carry_detection.py]: Because i have seen replays where the frame number is skipped by 1 or 2.
